### PR TITLE
8330427: Obsolete -XX:+PreserveAllAnnotations

### PR DESCRIPTION
**Pull request by alexmenkov**
Obsolete PreserveAllAnnotations flag which was deprecated in JDK 23.

Testing: tier1,tier2,tier3,tier4,hs-tier5-svc

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8330427](https://bugs.openjdk.org/browse/JDK-8330427): Obsolete -XX:+PreserveAllAnnotations (**Enhancement** - P4)

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) 🔄 Re-review required (review applies to [03aa9a76](https://git.openjdk.org/jdk/pull/20315/files/03aa9a7634c7ec9145b7b0c20a761a76628366c6))
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20315/head:pull/20315` \
`$ git checkout pull/20315`

Update a local copy of the PR: \
`$ git checkout pull/20315` \
`$ git pull https://git.openjdk.org/jdk.git pull/20315/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20315`

View PR using the GUI difftool: \
`$ git pr show -t 20315`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20315.diff">https://git.openjdk.org/jdk/pull/20315.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20315#issuecomment-2248669609)

### CODE REVIEW

**Review by coleenp — APPROVED** [review]
This looks really good. I hadn't expected so much code we had to preserve these.  Nice cleanup!

File: test/jdk/java/lang/instrument/RetransformRecordAnnotation.java [hunk 1]
```diff
@@ -29,7 +29,6 @@
  * @library /test/lib
  * @run shell MakeJAR.sh retransformAgent
  * @run main/othervm -javaagent:retransformAgent.jar -Xlog:redefine+class=trace RetransformRecordAnnotation
- * @run main/othervm -javaagent:retransformAgent.jar -XX:+PreserveAllAnnotations -Xlog:redefine+class=trace RetransformRecordAnnotation
```
Comment (dholmes-ora) [inline test/jdk/java/lang/instrument/RetransformRecordAnnotation.java:32]: This test is described as:
```
 * @summary test that records with invisible annotation can be retransformed
 ```
which suggests to me the test can actually be deleted as it serves no purpose now there are no invisible annotations

**Review by dholmes-ora — APPROVED** [review]
Great cleanup - good to see all that complexity go!

I think the test can be removed completely - see below.

Thanks

File: test/jdk/java/lang/instrument/RetransformRecordAnnotation.java (hunk 1 above)
Comment (alexmenkov) [reply test/jdk/java/lang/instrument/RetransformRecordAnnotation.java:32]: Agree.
Removed the test.

**Review by dholmes-ora — APPROVED** [review]

**Review by sspitsyn — APPROVED** [review]
Looks good. Really nice simplification.
Do I understand it right that all annotations are visible now, or we just do not parse/process invisible ones? If all annotations are visible then can we get rid of the suffix `_visible'?

**Comment by alexmenkov** [comment]
> Looks good. Really nice simplification. Do I understand it right that all annotations are visible now, or we just do not parse/process invisible ones? If all annotations are visible then can we get rid of the suffix `_visible'?

We skip (do not process) invisible annotations

**Comment by sspitsyn** [comment]
> We skip (do not process) invisible annotations
Okay, thanks.

**Pull request closed without merging**